### PR TITLE
Trim whitespace from process names

### PR DESCRIPTION
**Pull request by tomwilkie**
This causes detailed node lookups for the grouped-by-process-name view to fail.  Also, add a test for process walker trimmming whitespace, and a test the process-by-name view gives the right result.

Works around #281, but I believe there is an underlying UI bug where URLs aren't escaped properly.

### CODE REVIEW

**Comment by tomwilkie** [comment]
Paul's on holiday now.

File: app/api_topology_test.go [hunk 1]
```diff
@@ -43,14 +43,14 @@ func TestAPITopologyApplications(t *testing.T) {
...
+		body := getRawJSON(t, ts, fmt.Sprintf("/api/topology/applications/%s/%s", expected.ClientProcess1ID, expected.ServerProcessID))
 		var edge APIEdge
 		if err := json.Unmarshal(body, &edge); err != nil {
 			t.Fatalf("JSON parse error: %s", err)
 		}
 		want := render.AggregateMetadata{
-			"egress_bytes":  30,
-			"ingress_bytes": 300,
+			"egress_bytes":  10,
+			"ingress_bytes": 100,
```
Comment (peterbourgon) [inline app/api_topology_test.go]: Maybe it's still there and I'm not seeing it, but if not, can we add another test for something that would return the aggregate 30/300 bytes? Just want to make sure that functionality isn't overlooked.

**Comment by peterbourgon** [comment]
Otherwise LGTM.

File: app/api_topology_test.go (hunk 1 above)
Comment (tomwilkie) [inline app/api_topology_test.go]: Yeah a bunch of tests in render/topologies_test.go check for that on the the process by name, container and container images topologies.
